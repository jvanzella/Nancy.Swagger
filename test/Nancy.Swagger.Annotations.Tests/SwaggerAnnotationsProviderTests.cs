using ApprovalTests;
using ApprovalTests.Reporters;
using Nancy.Swagger.Modules;
using Nancy.Swagger.Services;
using Nancy.Testing;
using Xunit;

namespace Nancy.Swagger.Annotations.Tests
{
    [UseReporter(typeof(XUnitReporter))]
    public class SwaggerAnnotationsProviderTests
    {
        private readonly Browser _browser;

        public SwaggerAnnotationsProviderTests()
        {
            var bootstrapper = new ConfigurableBootstrapper(with =>
            {
                with.ApplicationStartup((container, pipelines) =>
                {
                    container.Register<ISwaggerMetadataProvider, SwaggerAnnotationsProvider>();
                });

                with.Module<SwaggerModule>();
                with.Module<TestRoutesModule>();
            });

            _browser = new Browser(bootstrapper);
        }

        [Fact]
        public void Get_ApiDocsRootpath_ReturnsResourceListing()
        {
            ApproveJsonResponse(_browser.Get("/api-docs"));
        }

        private static void ApproveJsonResponse(BrowserResponse response)
        {
            Approvals.VerifyJson(response.Body.AsString());
        }
    }
}