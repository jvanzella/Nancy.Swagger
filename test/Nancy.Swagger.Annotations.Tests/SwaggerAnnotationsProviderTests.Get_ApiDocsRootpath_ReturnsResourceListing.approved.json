{
    "swagger": "2.0",
    "info": {
        "title": "No title set",
        "version": "0.1"
    },
    "paths": {
        "/api-docs": {
            "get": {
                "summary": "Warning: no annotated method found for this route",
                "description": "[example]",
                "operationId": ""
            },
            "parameters": []
        },
        "/testroutes/anonymoushandler": {
            "get": {
                "summary": "Warning: no annotated method found for this route",
                "description": "[example]",
                "operationId": ""
            },
            "parameters": []
        },
        "/testroutes/not-annotated/get": {
            "get": {
                "summary": "Warning: no annotated method found for this route",
                "description": "[example]",
                "operationId": ""
            },
            "parameters": []
        },
        "/testroutes/strings": {
            "get": {
                "summary": "Some summary",
                "description": "Some notes",
                "operationId": "",
                "parameters": [],
                "responses": {}
            },
            "parameters": []
        },
        "/testroutes/strings/{id}": {
            "get": {
                "operationId": "",
                "parameters": [
                    {
                        "name": "id",
                        "in": "path",
                        "required": true
                    },
                    {
                        "name": "q",
                        "in": "query",
                        "description": "Query"
                    }
                ],
                "responses": {}
            },
            "parameters": []
        },
        "/testroutes/models": {
            "get": {
                "operationId": "",
                "parameters": [],
                "responses": {}
            },
            "parameters": []
        },
        "/testroutes/models/{id}": {
            "get": {
                "operationId": "",
                "parameters": [
                    {
                        "name": "id",
                        "in": "path"
                    }
                ],
                "responses": {}
            },
            "put": {
                "operationId": "",
                "parameters": [
                    {
                        "name": "testModel",
                        "in": "body"
                    }
                ],
                "responses": {}
            },
            "post": {
                "operationId": "",
                "parameters": [
                    {
                        "name": "testModel",
                        "in": "body",
                        "required": true
                    }
                ],
                "responses": {
                    "200": {
                        "description": "Everything OK"
                    },
                    "500": {
                        "description": "InternalServerError"
                    },
                    "404": {
                        "description": "NotFound"
                    },
                    "418": {
                        "description": "I'm a teapot"
                    }
                }
            },
            "delete": {
                "operationId": "",
                "parameters": [
                    {
                        "name": "id",
                        "in": "path"
                    }
                ],
                "responses": {}
            },
            "options": {
                "operationId": "",
                "parameters": [
                    {
                        "name": "id",
                        "in": "path"
                    }
                ],
                "responses": {}
            },
            "patch": {
                "operationId": "",
                "parameters": [
                    {
                        "name": "testModel",
                        "in": "body"
                    }
                ],
                "responses": {}
            },
            "parameters": []
        },
        "/testroutes/namedroute": {
            "get": {
                "operationId": "GetNamedRoute",
                "parameters": [],
                "responses": {}
            },
            "parameters": []
        },
        "/testroutes/allowmultipleparam": {
            "get": {
                "operationId": "",
                "parameters": [
                    {
                        "name": "ids",
                        "in": "query"
                    }
                ],
                "responses": {}
            },
            "parameters": []
        },
        "/testroutes/model-with-duplicate-typename": {
            "get": {
                "operationId": "",
                "parameters": [],
                "responses": {}
            },
            "parameters": []
        }
    }
}